mne/viz/_figure.py

```diff
@@ -341,11 +580,40 @@ def _click_ch_name(self, ch_index, button):
     def _resize_by_factor(self, factor):
         pass
 
+    @abstractmethod
+    def _get_ticklabels(self, orientation):
+        pass
+
+    @abstractmethod
+    def _update_yaxis_labels(self):
+        pass
+
 
 def _load_backend(backend_name):
     global backend
-    backend = importlib.import_module(name=_backends[backend_name],
-                                      package='mne.viz')
+    if backend_name == 'matplotlib':
+        backend = importlib.import_module(name='._mpl_figure',
+                                          package='mne.viz')
+    else:
+        repo_link = "git+https://github.com/mne-tools/" \
+                    "mne-qt-browser.git#egg=mne-qt-browser"
+        try:
+            from mne_qt_browser import _pg_figure as backend
+        except ModuleNotFoundError:
+            try:
+                subprocess.run(
+                    [sys.executable, "-m", "pip", "install",
+                     repo_link],
+                    check=True)
+            except subprocess.CalledProcessError:
+                print('Installation of pyqtgraph-backend failed!\n'
+                      f'You can try to install the pyqtgraph-backend '
+                      f'manually with:\n'
+                      f'"pip install {repo_link}"')
+                return _load_backend('matplotlib')
```

Comment:
@larsoner This automatic installation of mne-qt-browser won't be necessary anymore when it is put into `requirements.txt`, right?

Reply:
we need to discuss how to proceed here.

mne_qt_browser depends on mne and should take care of its release cycle and testing.

to me it's like mne-connectivity. We don't install or test mne-connectivity within mne-python repo.

anyway these lines of automatic install is really not recommended. Please remove them as otherwise we cannot merge and we would like to release in the coming days. This PR is to me the last big thing before we release.

thx @marsipu !

Reply:
Agreed no `pip install` should be done automatically

I think I agree with @agramfort that testing should be moved to the `mne-qt-browser` repo. @marsipu you can add a scheduled Azure run like we did for mne-kit-gui like [this](https://github.com/mne-tools/mne-kit-gui/blob/81fa55b64735ad0c6f35f9e4f4bfa5f78cf22589/azure-pipelines.yml#L13-L20) that runs every night, so you'll see quickly if things fail. I can also add an email notification via the Azure interface for you for the scheduled run -- I just did it for mne-kit-gui and it already pinged me over the weekend (somehow we managed to break things quickly!).

Reply:
@agramfort @larsoner Do you mean with moving testing to the `mne-qt-browser`-repo, that the changes from this PR in `test_raw.py` should be reverted and all the adapted tests should be copied to the `mne-qt-browser`?

Reply:
> Do you mean with moving testing to the mne-qt-browser-repo, that the changes from this PR in test_raw.py should be reverted and all the adapted tests should be copied to the mne-qt-browser?

Hmm, that does seem a bit tough. I guess for maintenance purposes, if we change something about how a MNEFigure class works for example, or if we want to test some new interaction mode, we would want to change mne-qt-browser at the same time. Not sure the best path here.

Thinking about it more I think I'm okay with merging this with the testing infrastructure as is. We can always split it off later / during the 1.0 cycle if we think of a better solution.

Thoughts @drammock @agramfort ?

Reply:
> Hmm, that does seem a bit tough. I guess for maintenance purposes, if we change something about how a MNEFigure class works for example, or if we want to test some new interaction mode, we would want to change mne-qt-browser at the same time.

I am glad, that's what I am thinking too, at least for the moment.

> you can add a scheduled Azure run like we did for mne-kit-gui like this that runs every night, so you'll see quickly if things fail

Is this setting up an Azure-pipeline something which has to be done before the release?

Reply:
> Is this setting up an Azure-pipeline something which has to be done before the release?

No, you can set it up on the `mne-qt-browser` repo anytime. I just created one for `mne-kit-gui` that was pretty simple and runs for all OSes, you might want to base things off of that if you want to take a stab at it.

Reply:
I thought that having this in a separate repo was temporary until all the kinks are worked out; if that's the case then we'll need the tests here eventually anyway.  Either way I am OK leaving the tests where they are for now and merging, but let's discuss at next dev meeting the best way to handle testing here.

Reply:
yes keep the "common test" in mne-python and make the mne-qt-browser CI run the test functions in mne-python. I am thinking something like what we use to test a new Raw object. We have a function and if this function passes for a backend then we are good.

makes sense?